lib/config/default-config.js

```diff
@@ -0,0 +1,48 @@
+/**
+ * @fileoverview Default configuration
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const Rules = require("../rules");
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+/*
+ * Because we try to delay loading rules until absolutely necessary, a proxy
+ * allows us to hook into the lazy-loading aspect of the rules map while still
+ * keeping all of the relevant configuration inside of the config array.
+ */
+const defaultRulesConfig = new Proxy({}, {
+    get(target, property) {
+        return Rules.get(property);
+    },
+
+    has(target, property) {
+        return Rules.has(property);
+    }
+});
+
+exports.defaultConfig = [
+    {
+        plugins: {
+            "@": {
+                parsers: {
+                    espree: require("espree")
+                },
+                rules: defaultRulesConfig
+            }
+        },
+        ignores: ["node_modules/**"],
```

Comment:
These are all glob patterns now. I’ll update them to match the current defaults.

Reply:
Does it also mean that we'll interpret the content of `.eslintignore` file differently, depending on the format of the found eslint configuration file: 

* as a list of `.gitignore` patterns (actual behavior) if the config is `.eslintrc.*`.
* as a list of glob patterns if the config is `eslint.config.js`

Reply:
No. There is no change to how .eslintignore will be interpreted. It will be read and compiled into a function that is then included in a flat config ignores key. (Can be either a glob pattern or a function.)